collector/pg_stat_bgwriter.go

```diff
@@ -0,0 +1,212 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector("bgwriter", defaultEnabled, NewPGStatBGWriterCollector)
+}
+
+type PGStatBGWriterCollector struct {
+}
+
+func NewPGStatBGWriterCollector(logger log.Logger) (Collector, error) {
+	return &PGStatBGWriterCollector{}, nil
+}
+
+const bgWriterSubsystem = "stat_bgwriter"
+
+var statBGWriter = map[string]*prometheus.Desc{
+	"checkpoints_timed": prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, bgWriterSubsystem, "checkpoints_timed_total"),
+		"Number of scheduled checkpoints that have been performed",
+		[]string{"server"},
```

Comment:
I just realized I think we have something going wrong with these labels. This seems like it's a discovery label. Each scrape should only ever have one `server` in the output.

Reply:
The DSNs passed in can be multiple so this would be the only way to distinguish metrics between them.

Reply:
Ugh, this exporter is so broken. :-(